File: solutions/solutions.kt
fun gcd(a: Long, b: Long): Long {
    if (a == 0L) {
        return b
    }

    return gcd(b % a, a)
}

fun main() {
    val t = readln().toInt()

    for (i in 0 until t) {
        val (n, a, b) = readln().split(' ').map { it.toLong() }

        if (a == 0L || b == 0L) {
            println(2)
        } else if (n == a + b) {
            println(3)
        } else {
            println(n/gcd(n, gcd(a, b)))
        }
    }
}

File: solutions/solutions.py
from sys import stdin
from math import gcd

T = int(stdin.readline())

for t in range(T):
    N, a, b = [int(i) for i in stdin.readline().strip().split(' ')]

    if a == 0 or b == 0:
        print(2)
    elif N == a + b:
        print(3)
    else:
        print(N//gcd(N, gcd(a, b)))
